packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -198,19 +198,60 @@ export async function uploadFile(node: ZoweDatasetNode, doc: vscode.TextDocument
  * @param {IZoweDatasetTreeNode} node - The node selected for deletion
  * @param {DatasetTree} datasetProvider - the tree which contains the nodes
  */
-export async function deleteDatasetPrompt(datasetProvider: IZoweTree<IZoweDatasetTreeNode>) {
+export async function deleteDatasetPrompt(
+    datasetProvider: IZoweTree<IZoweDatasetTreeNode>,
+    node?: IZoweDatasetTreeNode
+) {
+    let nodes: IZoweDatasetTreeNode[];
     const treeView = datasetProvider.getTreeView();
     const selectedNodes: IZoweDatasetTreeNode[] = treeView.selection;
+    let includedSelection = false;
+    if (node) {
+        for (const item of selectedNodes) {
+            if (node.getLabel() === item.getLabel()) {
+                includedSelection = true;
+            }
+        }
+    }
+    if (includedSelection || !node) {
+        // Filter out sessions, favorite nodes, or information messages
+        nodes = selectedNodes.filter(
+            (selectedNode) =>
+                selectedNode.getParent() &&
+                !contextually.isFavorite(selectedNode) &&
+                !contextually.isFavorite(selectedNode.getParent()) &&
+                !contextually.isSession(selectedNode) &&
+                !contextually.isInformation(selectedNode)
+        );
+        // Check that child and parent aren't both in array
+        const newNodeList: IZoweDatasetTreeNode[] = [];
+        for (const item of nodes) {
+            if (contextually.isDsMember(item)) {
```

Comment:
I the problem is here missing the `!`: Shouldn't it be `if (!contextually.isDsMember(item)) {`